docs/developer-docs/latest/development/typescript.md

```diff
@@ -4,110 +4,18 @@ description: Learn how you can use Typescript for your Strapi application.
-| `./dist` directory                        | Adds the location for compiling the project JavaScript source code.                                                                               | application root |
-| `build` directory                         | Contains the compiled administration panel JavaScript source code.  The directory is created on the first `yarn build` or `npm run build` command | `./dist/`         |
-| `tsconfig.json` file                      | Manages TypeScript compilation for the server.                                                                                                    | application root |
-| `tsconfig.json` file                      | Manages TypeScript compilation for the admin panel.                                                                                               | `./src/admin/`   |
-
-Starting the development environment for a TypeScript-enabled project requires building the admin panel prior to starting the server. In development mode, the application source code is compiled to the `./dist/build` directory and recompiled with each change in the Content-type Builder. To start the application, run the following commands in the root directory:
-
-<code-group>
-
-<code-block title="NPM">
-
-```sh
-npm run build
-npm run develop
-```
-
-</code-block>
-
- <code-block title="YARN">
-
-```sh
-yarn build
-yarn develop
-```
-
-</code-block>
-
-</code-group>
-
-## Use TypeScript typings
-
-Strapi provides typings on the Strapi class to improve the TypeScript experience. In the following example a `register` method is added to `./src/index.ts`. For example, when the `strapi.runLifecyclesFunctions` method is added, a list of available lifecycle types (`register`, `bootstrap` and `destroy`) are returned by the code editor. The following example is illustrated statically and the second tab shows an animated example.
-
-:::: tabs card
-
-::: tab STATIC EXAMPLE
-
-```ts
-// path: ./src/index.ts
-
-import '@strapi/strapi';
-
-export default {
-
-  register( {strapi }: { strapi: Strapi }) {
-    strapi.runLifecyclesFunctions('register') .then(()  => console.log('hello world')
-    )
-    
-  },
-};
-```
-
-:::
-
-::::
-
+TypeScript adds an additional type system layer above JavaScript, which means that existing JavaScript code is also TypeScript code. Strapi supports TypeScript in new projects on v4.2.0 and above. To start developing in TypeScript use the [CLI documentation](/developer-docs/latest/setup-deployment-guides/installation/cli.md) to create a new TypeScript project. Additionally, the [project structure](/developer-docs/latest/setup-deployment-guides/file-structure.md) and [TypeScript configuration](/developer-docs/latest/setup-deployment-guides/configurations/optional/typescript.md) sections have TypeScript-specific resources for understanding and configuring an application. The Developer Documentation contains code examples in both JavaScript and TypeScript.
```

Comment:
```suggestion
TypeScript adds an additional type system layer above JavaScript, which means that existing JavaScript code is also TypeScript code. Strapi supports TypeScript in new projects on v4.2.0 and above. TypeScript-enabled projects allow developing plugins with TypeScript as well as using TypeScript typings.

::: strapi Getting started with TypeScript
To start developing in TypeScript, use the [CLI documentation](/developer-docs/latest/setup-deployment-guides/installation/cli.md) to create a new TypeScript project. Additionally, the [project structure](/developer-docs/latest/setup-deployment-guides/file-structure.md) and [TypeScript configuration](/developer-docs/latest/setup-deployment-guides/configurations/optional/typescript.md) sections have TypeScript-specific resources for understanding and configuring an application.
:::
```

I let your rephrase the sentence I added if it's not good enough (and check it works well), it was mainly to give an idea of what I had in mind. I thought the introduction felt like unrelated sentences put together, so in my opinion it could help to focus just on what TypeScript is and what the documentation is here for, and then use the Strapi callout to mention other links. I'm aware that for now the documentation is short and it might look a bit busy with all the callouts but it should look better later once there's more content in the page and we remove the first callout.

Reply:
ok thanks